### ISSUE TITLE
如何判断一个样本是unknow？

### ISSUE THREAD

**Issue by liyangfan0** [00why00/Glocal#12]
作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。

**Comment by jiajingchen113322** [comment]
> 作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。

I have the same question for the author. During the evaluation, author used the first 75 samples as unknown, that means author make use of the prior information that half the query is unknown and the rest are known. However, in the real application, we don't know how many unknown and known class objects are in the query.

**Comment by liyangfan0** [comment]
> > 作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。
> 
> I have the same question for the author. During the evaluation, author used the first 75 samples as unknown, that means author make use of the prior information that half the query is unknown and the rest are known. However, in the real application, we don't know how many unknown and known class objects are in the query.

**Comment by liyangfan0** [comment]
> 作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。

**Comment by liyangfan0** [comment]
> > 作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。
> 
> I have the same question for the author. During the evaluation, author used the first 75 samples as unknown, that means author make use of the prior information that half the query is unknown and the rest are known. However, in the real application, we don't know how many unknown and known class objects are in the query.

I agree with you. I have given up following this paper because I think it is unreasonable and unrealistic.

**Comment by LoveMiki** [comment]
> > 作者你好！请教一个问题，在测试的时候论文是将前75个有着最大energy score的query作为unknow，但如果是测试一张query，如何判断它是unknow呢？很多论文是采用阈值来判断unknow，而这里直接用前75个作为unknow，实际上引入了先验信息，这种测试方式我很疑惑到底合不合理。
> 
> I have the same question for the author. During the evaluation, author used the first 75 samples as unknown, that means author make use of the prior information that half the query is unknown and the rest are known. However, in the real application, we don't know how many unknown and known class objects are in the query.

Agreed. I also found that the author used the meta-trained model of FEAT to further meta-train GEL, which is not reasonable and unfair to other methods.

**Comment by 00why00** [comment]
Sorry for the late response. I'll answer your questions one by one.

**1. How to judge unknown samples during the evaluation**
In order to ensure the fairness of the comparison, our judgment method is consistent with [SnaTCHer](https://github.com/MinkiJ/SnaTCHer), [TANE](https://github.com/shiyuanh/TANE), [RFDNet](https://github.com/shule-deng/RFDNet) et al., that is, TPR and FPR are calculated by the open label of each sample to calculate the AUROC.

Ours:
https://github.com/00why00/Glocal/blob/db161dbf1e80fabfccd96cd323bf485e50f70e1c/model/utils/train_utils.py#L70-L80

SnaTCHer:
https://github.com/MinkiJ/SnaTCHer/blob/335c42469f0a7ad72c5c3480c8effc8c293823e0/model/trainer/fsl_trainer_SnaTCHerF.py#L17-L23

TANE:
https://github.com/shiyuanh/TANE/blob/fe089d477ba9387dd2d008d8872cdc84f8ad77ed/trainer/FSEval.py#L53-L73

RFDNet:
https://github.com/shule-deng/RFDNet/blob/bc03309447b9013415563fdc6f9a938a0443d525/train.py#L206-L209

**2. Why use the meta-trained model of FEAT to further meta-train GEL**
For a fair comparison, we trained the model in the same way as our baseline [SnaTCHer](https://github.com/MinkiJ/SnaTCHer), i.e. use the meta-trained model of FEAT to further meta-train our model.
